### PR TITLE
applications: machine_learning: Add Thingy:53 configuration

### PR DESCRIPTION
**Pull request by emob-nordic**
This change adds Thingy:53 board configuration files for machine learning app.

JIRA: NCSDK-9476

### CODE REVIEW

File: applications/machine_learning/CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,9 @@ set(mcuboot_CONF_FILE
 set(b0_CONF_FILE
   ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/b0_${CMAKE_BUILD_TYPE}.conf
   )
+set(hci_rpmsg_CONF_FILE
+  ${CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/child_image/hci_rpmsg_${CMAKE_BUILD_TYPE}.conf
```
Comment (MarekPieta) [inline applications/machine_learning/CMakeLists.txt:38]: I think we could define path as:   `$CMAKE_CURRENT_SOURCE_DIR}/configuration/${BOARD}/hci_rpmsg_${CMAKE_BUILD_TYPE}.conf`. That would be more consistent with our way of defining configuration for bootloaders.

File: subsys/caf/modules/ble_state.c [hunk 2]
```diff
@@ -83,7 +83,11 @@ static void set_tx_power(struct bt_conn *conn)
 		cp = net_buf_add(buf, sizeof(*cp));
 		cp->handle = sys_cpu_to_le16(conn_handle);
 		cp->handle_type = BT_HCI_VS_LL_HANDLE_TYPE_CONN;
+#ifndef CONFIG_BT_CTLR
```
Comment (MarekPieta) [inline subsys/caf/modules/ble_state.c:86]: Consider using the following syntax at the top of the file
```
#ifdef CONFIG_..
   #define TX_POWER CONFIG__...
#else
   #define TX_POWER 0
#endif
```

File: applications/machine_learning/CMakeLists.txt (hunk 1 above)
Comment (emob-nordic) [reply applications/machine_learning/CMakeLists.txt:38]: No strong opinion but NCS users may be used to having `child_image` directory for other cores. The thing you describe are child images for same core, which are then merged into single hex. @pdunaj what is your opinion on this?

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app_ZDebug.conf [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+CONFIG_GPIO=y
+
+CONFIG_LED=y
+CONFIG_LED_GPIO=y
+
+# Using built-in sensor
+CONFIG_SENSOR=y
+CONFIG_ADXL362=y
+CONFIG_ADXL362_ACCEL_RANGE_8G=y
+CONFIG_ADXL362_ACCEL_ODR_200=y
```
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app_ZDebug.conf:91]: 100 Hz may be sufficient in our case (we sample every 20 ms - 50Hz)

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/child_image/hci_rpmsg_ZDebug.conf [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+CONFIG_RPMSG_SERVICE=y
+CONFIG_RPMSG_SERVICE_MODE_REMOTE=y
+
+CONFIG_HEAP_MEM_POOL_SIZE=8192
+
+CONFIG_MAIN_STACK_SIZE=512
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=512
+CONFIG_BT=y
+CONFIG_BT_HCI_RAW=y
+CONFIG_BT_MAX_CONN=16
```
Comment (MarekPieta) [inline applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/child_image/hci_rpmsg_ZDebug.conf:18]: Consider setting it to `1` to be consistent with application configuration requirements. CAF BLE implementation assumes that Bluetooth peripheral has only 1 simultaneous connection.

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app_ZDebug.conf (hunk 3 above)
Comment (emob-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app_ZDebug.conf:91]: leftover, right

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/child_image/hci_rpmsg_ZDebug.conf (hunk 4 above)
Comment (emob-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/child_image/hci_rpmsg_ZDebug.conf:18]: Ok.

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app_ZDebug.conf [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+CONFIG_CAF_CLICK_DETECTOR=y
+
+CONFIG_CAF_BLE_STATE=y
+CONFIG_CAF_BLE_USE_LLPM=n
+
+CONFIG_CAF_BLE_ADV=y
+CONFIG_CAF_BLE_ADV_FAST_ADV=y
+
+CONFIG_CAF_LEDS=y
+CONFIG_CAF_LEDS_GPIO=y
```
[the remark this answers is not in the document]
Comment (emob-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/app_ZDebug.conf:43]: As you probably see i copy pasted that, will apply your suggestions. Thanks!

File: applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/child_image/hci_rpmsg_ZDebug.conf (hunk 4 above)
Comment (emob-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/child_image/hci_rpmsg_ZDebug.conf:18]: 2 is minimal and default value, cant get lower due to Kconfig dependencies.

Comment (emob-nordic) [reply applications/machine_learning/configuration/thingy53_nrf5340_cpuapp/child_image/hci_rpmsg_ZDebug.conf:18]: I would just remove that line.

File: applications/machine_learning/CMakeLists.txt (hunk 1 above)
Comment (emob-nordic) [reply applications/machine_learning/CMakeLists.txt]: Talked offline with @pdunaj, changed as proposed.

**Review by MarekPieta — APPROVED** [review]

**Review by pdunaj — APPROVED** [review]

**Pull request merged**